INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:15:22] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:15:22] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:19:50] "[31m[1mPOST /api/v1/login HTTP/1.1[0m" 401 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:20:04] "[31m[1mPOST /api/v1/login HTTP/1.1[0m" 400 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/login_controller.py", line 10, in login
    return fun_login()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py", line 12, in fun_login
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:20:12] "[31m[1mPOST /api/v1/login HTTP/1.1[0m" 415 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:22:41] "POST /api/v1/admin/createUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:22:58] "POST /api/v1/admin/createUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:23:53] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:23:53] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:25:12] "POST /api/v1/updatepwd HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:25:29] "POST /api/v1/accountRecovery HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:25:51] "GET /api/v1/verifyOTP HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:26:51] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:26:51] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:27:34] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:27:34] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:27:39] "[31m[1mPOST /api/v1/admin/createUser HTTP/1.1[0m" 401 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:28:09] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:28:09] "GET /api/v1/mentor/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:28:27] "[35m[1mPOST /api/v1/mentor/projects/addStudent HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:28:49] "POST /api/v1/mentor/addAchievement HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:29:45] "POST /api/v1/mentor/addAchievement HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:07] "POST /api/v1/mentor/projects/addStudent HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:15] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:15] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:23] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:23] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:32] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:39] "GET /api/v1/student/project HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:30:56] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:32:14] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:32:14] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:32:22] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Feb/2024 14:32:22] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:10:39] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 415 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:11:28] "GET /api/v1/mentor/projects/students HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:11:31] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 415 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:12:46] "[35m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:13:52] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 415 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:14:20] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 415 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/run.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:15:04] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 415 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:41:41] "GET /api/v1/mentor/projects/students HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:53:35] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 415 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 10:53:47] "GET /api/v1/mentor/projects/students HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 614, in get_json
    rv = self.json_module.loads(data)
  File "/usr/local/lib/python3.8/dist-packages/flask/json/provider.py", line 188, in loads
    return json.loads(s, **kwargs)
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 3 column 1 (char 29)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 623, in get_json
    rv = self.on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 648, in on_json_loading_failed
    raise BadRequest(f"Failed to decode JSON object: {e}")
werkzeug.exceptions.BadRequest: 400 Bad Request: Failed to decode JSON object: Expecting property name enclosed in double quotes: line 3 column 1 (char 29)
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:26:18] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 400 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 614, in get_json
    rv = self.json_module.loads(data)
  File "/usr/local/lib/python3.8/dist-packages/flask/json/provider.py", line 188, in loads
    return json.loads(s, **kwargs)
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 3 column 1 (char 21)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 55, in project_students
    students = get_project_students()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 56, in get_project_students
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 623, in get_json
    rv = self.on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 648, in on_json_loading_failed
    raise BadRequest(f"Failed to decode JSON object: {e}")
werkzeug.exceptions.BadRequest: 400 Bad Request: Failed to decode JSON object: Expecting value: line 3 column 1 (char 21)
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:26:27] "[31m[1mGET /api/v1/mentor/projects/students HTTP/1.1[0m" 400 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:26:34] "GET /api/v1/mentor/projects/students HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:26:55] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:26:55] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:29:37] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:29:37] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:33:02] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:33:02] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:34:02] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/admin_controller.py", line 46, in admin_home
    return fun_admin_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py", line 28, in fun_admin_home
    return jsonify({{'Type':'role_id'}:{'First_Name' : f'{admin[0]}'}})
TypeError: unhashable type: 'dict'
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:34:02] "[35m[1mGET /api/v1/admin/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:36:29] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:36:29] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:45:05] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:45:05] "GET /api/v1/admin/home HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/login_controller.py", line 10, in login
    return fun_login()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py", line 12, in fun_login
    data = request.json
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 561, in json
    return self.get_json()
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 607, in get_json
    return self.on_json_loading_failed(None)
  File "/usr/local/lib/python3.8/dist-packages/flask/wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/wrappers/request.py", line 650, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:45:13] "[31m[1mPOST /api/v1/login HTTP/1.1[0m" 415 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:45:18] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:45:18] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:58:38] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/admin_controller.py", line 46, in admin_home
    return fun_admin_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py", line 30, in fun_admin_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:58:38] "[35m[1mGET /api/v1/admin/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:59:35] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/admin_controller.py", line 46, in admin_home
    return fun_admin_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py", line 30, in fun_admin_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:59:35] "[35m[1mGET /api/v1/admin/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:59:48] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/admin_controller.py", line 46, in admin_home
    return fun_admin_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py", line 30, in fun_admin_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 14:59:48] "[35m[1mGET /api/v1/admin/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:00:08] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/admin_controller.py", line 46, in admin_home
    return fun_admin_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py", line 30, in fun_admin_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:00:08] "[35m[1mGET /api/v1/admin/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:00:51] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/admin_controller.py", line 46, in admin_home
    return fun_admin_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py", line 30, in fun_admin_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:00:51] "[35m[1mGET /api/v1/admin/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:01:52] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:01:52] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:02:07] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:02:07] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:03:47] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:03:47] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:05:05] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 15:05:05] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 20:59:12] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 20:59:12] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:10:58] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 31, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 32, in fun_student_home
    except psycopg2.error as e:
AttributeError: module 'psycopg2' has no attribute 'error'
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:10:58] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:12:14] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 32, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 33, in fun_student_home
    except psycopg2.error as e:
AttributeError: module 'psycopg2' has no attribute 'error'
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:12:14] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:12:34] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 32, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 33, in fun_student_home
    except psycopg2.errors as e:
TypeError: catching classes that do not inherit from BaseException is not allowed
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:12:34] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:13:20] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 32, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:13:21] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:14:09] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 32, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:14:09] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:18:16] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 32, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:18:16] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:19:04] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 43, in fun_student_home
    'department':dept,
NameError: name 'dept' is not defined
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:19:04] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:21:02] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 30, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:21:02] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:50:48] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 33, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:50:48] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:52:41] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:52:41] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:54:16] "[31m[1mPOST /api/v1/login HTTP/1.1[0m" 401 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:55:15] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 35, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 21:55:15] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:34:51] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:34:51] "[33mGET /api/v1/student/home HTTP/1.1[0m" 404 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:35:23] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:35:23] "[33mGET /api/v1/student/home HTTP/1.1[0m" 404 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:36:44] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 33, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:36:44] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:48:21] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 34, in fun_student_home
    if dept is None: return jsonify({'error': f'Error fetching department for student {st_id}: {e}'}), 404
UnboundLocalError: local variable 'e' referenced before assignment
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:48:21] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:49:22] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:49:22] "[33mGET /api/v1/student/home HTTP/1.1[0m" 404 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:55:13] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:55:13] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:55:42] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:55:42] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:58:12] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:58:12] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:58:30] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 33, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:58:30] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:58:55] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 33, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:58:55] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:59:22] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 11, in student_home
    return fun_student_home()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 33, in fun_student_home
    dept = cursor.fetchone()[0]
TypeError: 'NoneType' object is not subscriptable
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 22:59:22] "[35m[1mGET /api/v1/student/home HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:00:31] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:00:31] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:02:18] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:02:18] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:02:27] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:02:27] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:06:55] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:06:55] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:07:07] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:07:55] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:07:55] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:08:00] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:08:05] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 16, in update_student
    return fun_update_student()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 80, in fun_update_student
    if not is_valid_phone(update_fields['sphoneno']):
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/validation_utils.py", line 12, in is_valid_phone
    return bool(re.match(pattern, phone))
  File "/usr/lib/python3.8/re.py", line 191, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or bytes-like object
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:08:12] "[35m[1mPUT /api/v1/student/editProfile HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:08:26] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [07/Feb/2024 23:08:30] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:50:41] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:50:41] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:50:48] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:50:52] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:51:06] "[31m[1mGET /api/v1/student/project HTTP/1.1[0m" 401 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:51:11] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:51:15] "[31m[1mGET /api/v1/student/project HTTP/1.1[0m" 401 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:52:48] "[31m[1mGET /api/v1/student/project HTTP/1.1[0m" 401 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:52:53] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:52:58] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:52:58] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/Feb/2024 11:53:02] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:25:43] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:25:43] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:25:50] "[31m[1mGET /api/v1/student/project HTTP/1.1[0m" 401 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:25:54] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:25:58] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:08] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:08] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:23] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:26] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:30] "GET /api/v1/student/project HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:33] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:52] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:40:52] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:41:04] "POST /api/v1/admin/createUser HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:55:11] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:55:11] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:55:17] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:55:17] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:55:26] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:56:13] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:56:13] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:57:17] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 09:59:24] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 21, in view_student_achievements
    return fun_view_student_achievements()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 122, in fun_view_student_achievements
    achievements_list = {
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 123, in <setcomp>
    {'achievement_id': row[0], 'achievement_name': row[1], 'description': row[2], 'points': row[3]}
TypeError: unhashable type: 'dict'
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:01:35] "[35m[1mGET /api/v1/student/achievements HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 21, in view_student_achievements
    return fun_view_student_achievements()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 128, in fun_view_student_achievements
    achievements_dict = {
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 129, in <dictcomp>
    row[0]: {'achievement_name': row[1], 'description': row[2], 'points': row[3]}
KeyError: 0
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:34:26] "[35m[1mGET /api/v1/student/achievements HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:34:32] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:34:33] "GET /api/v1/student/home HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 21, in view_student_achievements
    return fun_view_student_achievements()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 128, in fun_view_student_achievements
    achievements_dict = {
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 129, in <dictcomp>
    row[0]: {'achievement_name': row[1], 'description': row[2], 'points': row[3]}
KeyError: 0
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:34:36] "[35m[1mGET /api/v1/student/achievements HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 21, in view_student_achievements
    return fun_view_student_achievements()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 128, in fun_view_student_achievements
    achievements_dict = {
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 129, in <dictcomp>
    row[0]: {'achievement_name': row[1], 'description': row[2], 'points': row[3]}
KeyError: 0
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:35:59] "[35m[1mGET /api/v1/student/achievements HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 21, in view_student_achievements
    return fun_view_student_achievements()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 128, in fun_view_student_achievements
    achievements_dict = {
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 129, in <dictcomp>
    row[0]: {'achievement_name': row[1], 'description': row[2], 'points': row[3]}
KeyError: 0
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:36:39] "[35m[1mGET /api/v1/student/achievements HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 21, in view_student_achievements
    return fun_view_student_achievements()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 128, in fun_view_student_achievements
    achievements_dict = {
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 129, in <dictcomp>
    row[0]: {'achievement_name': row[1], 'description': row[2], 'points': row[3]}
KeyError: 0
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:36:59] "[35m[1mGET /api/v1/student/achievements HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:37:07] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/student_controller.py", line 21, in view_student_achievements
    return fun_view_student_achievements()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 128, in fun_view_student_achievements
    achievements_dict = {
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py", line 129, in <dictcomp>
    row[0]: {'achievement_name': row[1], 'description': row[2], 'points': row[3]}
KeyError: 0
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:37:12] "[35m[1mGET /api/v1/student/achievements HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:37:41] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:42:03] "GET /api/v1/student/project HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:57:08] "GET /api/v1/student/project HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:58:29] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:58:33] "GET /api/v1/student/project HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 10:58:36] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:10:01] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:15:43] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:16:05] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:18:41] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:19:41] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:22:31] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:23:06] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:23:54] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:26:06] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:34:20] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:34:56] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:35:31] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:55:24] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:56:12] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:56:14] "GET /api/v1/student/progress HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:56:22] "POST /api/v1/admin/createUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 15:56:27] "GET /api/v1/mentor/projects HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:04] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:04] "GET /api/v1/student/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:09] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:09] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:15] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:15] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:20] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:23] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:28] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:03:58] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:05:02] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:05:04] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:05:10] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:05:10] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:05:13] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:05:47] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:05:47] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:06:00] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:06:00] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:06:04] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:06:16] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:06:44] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:06:44] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:06:48] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:07:40] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:07:45] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:07:49] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:07:49] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:07:52] "GET /api/v1/student/project HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/session_manager.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:09:18] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:09:18] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:09:22] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/student.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:12:10] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:12:10] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:12:24] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:12:24] "GET /api/v1/student/home HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/admin_controller.py", line 25, in admin_update_user
    return fun_admin_update_user()
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py", line 222, in fun_admin_update_user
    if not check_login(role_type): return generate_response('Unauthorized access! Please login first',401)
UnboundLocalError: local variable 'role_type' referenced before assignment
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:12:33] "[35m[1mPOST /api/v1/admin/updateUser HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:14:28] "POST /api/v1/admin/updateUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:14:33] "POST /api/v1/admin/createUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:14:45] "DELETE /api/v1/admin/removeUser?username=USRX241016 HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:36:56] "POST /api/v1/admin/updateUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:09] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:09] "GET /api/v1/mentor/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:14] "POST /api/v1/admin/createUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:21] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:26] "GET /api/v1/mentor/projects/students HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:32] "GET /api/v1/mentor/projects HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:59] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:39:59] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:40:05] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:40:23] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:42:10] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:42:12] "GET /api/v1/student/achievements HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:42:16] "PUT /api/v1/student/editProfile HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:52:21] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:52:21] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:52:28] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 16:52:28] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:00:55] "POST /api/v1/updatepwd HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:00:59] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:00:59] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:01:39] "POST /api/v1/updatepwd HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:03:13] "POST /api/v1/updatepwd HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:04:01] "[31m[1mPOST /api/v1/updatepwd HTTP/1.1[0m" 401 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:04:04] "POST /api/v1/updatepwd HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/login.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/admin.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:15:39] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:15:39] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:15:45] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:15:45] "GET /api/v1/admin/home HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:15:59] "POST /api/v1/admin/createUser HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:18:02] "[32mPOST /api/v1/logout HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:18:02] "GET /api/v1/temp HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:19:24] "[32mPOST /api/v1/login HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:19:25] "GET /api/v1/mentor/home HTTP/1.1" 200 -
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py", line 92, in format_query_results
    with psycopg2.connect() as conn:
  File "/home/geojoseph/.local/lib/python3.8/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "geojoseph" does not exist


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 41, in mentor_projects
    return get_mentor_projects(username)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 52, in get_mentor_projects
    response=format_query_results(projects)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py", line 106, in format_query_results
    return generate_response(error=error_message, status_code=500)
TypeError: generate_response() got an unexpected keyword argument 'error'
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:19:45] "[35m[1mGET /api/v1/mentor/projects HTTP/1.1[0m" 500 -
INFO:werkzeug: * Detected change in '/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py', reloading
INFO:werkzeug: * Restarting with stat
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 379-541-606
ERROR:root:An error occurred:
Traceback (most recent call last):
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py", line 92, in format_query_results
    with psycopg2.connect() as conn:
  File "/home/geojoseph/.local/lib/python3.8/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "geojoseph" does not exist


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/controller/mentor_contoller.py", line 41, in mentor_projects
    return get_mentor_projects(username)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/services/mentor.py", line 52, in get_mentor_projects
    response=format_query_results(projects)
  File "/home/geojoseph/Desktop/Mini Project/userprofilemanagement/app/utils/response_utils.py", line 106, in format_query_results
    return generate_response(error=error_message, status_code=500)
TypeError: generate_response() got an unexpected keyword argument 'error'
INFO:werkzeug:127.0.0.1 - - [09/Feb/2024 17:20:26] "[35m[1mGET /api/v1/mentor/projects HTTP/1.1[0m" 500 -
